### PR TITLE
[bufferorch] : Support for buffer profiles for VoQ on chassis (#2465)

### PR DESCRIPTION
**Pull request by vmittal-msft**
* QoS changes for VoQ on chassis

**What I did**
Porting https://github.com/sonic-net/sonic-swss/pull/2465 from master branch

**Why I did it**
Due to merge conflict

**How I verified it**

**Details if related**

### CODE REVIEW

**Comment by gechiang** [comment]
@vmittal-msft can you please resolve the conflict?

**Comment by vmittal-msft** [comment]
@skbarista please reiview.

File: orchagent/portsorch.cpp [hunk 1]
```diff
@@ -7786,13 +7786,14 @@ bool PortsOrch::addSystemPorts()
                 }
 
                 //System port for local port. Update the system port info in the existing physical port
-                if(!getPort(attr.value.oid, port))
```
Comment (skbarista) [inline orchagent/portsorch.cpp:7789]: @vmittal-msft We should leave this code untouched.

File: orchagent/portsorch.cpp [hunk 2]
```diff
@@ -6101,7 +6101,7 @@ void PortsOrch::generateQueueMapPerPort(const Port& port, FlexCounterQueueStates
         {
 	    /* voq counters are always enabled. There is no mechanism to disable voq
 	     * counters in a voq system. */
-            if (!voq && !queuesState.isQueueCounterEnabled(queueRealIndex))
+            if (!(gMySwitchType == "voq") && !queuesState.isQueueCounterEnabled(queueRealIndex))
```
Comment (skbarista) [inline orchagent/portsorch.cpp:6104]: Instead of !(gMySwitchType == "voq")  maybe just change  the check to ( gMySwitchType != "voq" )

**Comment by vmittal-msft** [comment]
dependent on https://github.com/sonic-net/sonic-swss/pull/2685

**Comment by gechiang** [comment]
> dependent on #2685

This PR has a dependency on which 2685?  SWSS?  or other submodule?

**Review by prsunny — APPROVED** [review]

**Pull request merged**